custom_components/eo_mini/switch.py

```diff
@@ -69,3 +72,144 @@ async def async_turn_off(self, **kwargs):
     def unique_id(self):
         "Return a unique ID to use for this entity."
         return f"{DOMAIN}_charger_{self.coordinator.serial}_locked"
+
+
+class EOMiniOffPeakSwitch(EOMiniChargerEntity, SwitchEntity):
+    "Switch entity to represent the enabled/disabled status of Off-Peak charging"
+    coordinator: EODataUpdateCoordinator
+
+    def __init__(self, *args):
+        self.entity_description = SwitchEntityDescription(
+            key="Off_Peak_Charging",
+            name="Off Peak Charging",
+            device_class=SwitchDeviceClass.SWITCH,
+            icon="mdi:sort-clock-ascending-outline",
+        )
+        super().__init__(*args)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        "Handle updated data from the coordinator."
+        if self.coordinator.charge_options:
```

Comment:
Is this ever None? Or are we being defensive here?

Reply:
I was copying the pattern you had with the `EOMiniLockSwitch` from line 48.

I'll get the other changes above sorted and tested over the next day or so

Reply:
Makes sense, thanks!